From: Don Skidmore <donald.c.skidmore@intel.com>
Date: Mon, 11 Jul 2016 21:29:56 -0400
Subject: ixgbevf: Add lock around ixgbevf_reinit_locked call
Patch-mainline: v4.8-rc1
Git-commit: 8e8247ab98315e096b812a68381ca812f0b0b5e3
References: bsc#1024349 FATE#321240 bsc#1024355 FATE#321241

The function ixgbevf_reinit_locked() assumes you have the rtnl lock
however we didn't when calling from the service task.  This patch
corrects that.

Signed-off-by: Don Skidmore <donald.c.skidmore@intel.com>
Tested-by: Andrew Bowers <andrewx.bowers@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c
+++ b/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c
@@ -2775,7 +2775,9 @@ static void ixgbevf_reset_subtask(struct
 
 	adapter->tx_timeout_count++;
 
+	rtnl_lock();
 	ixgbevf_reinit_locked(adapter);
+	rtnl_unlock();
 }
 
 /**
